qbraid/runtime/result.py

```diff
@@ -268,6 +268,55 @@ def to_dict(self) -> dict[str, Any]:
         }
 
 
+class AnnealingResultData(ResultData):
+    """Class for storing and accessing the results of an annealing job.
+
+    Args:
+        ResultData (ResultData): ABC for runtime results linked to a specific ExperimentType.
+
+    Returns:
+        _type_: _description_
+    """
+
+    def __init__(self, job_data: Optional[dict[str, Any]] = None, **kwargs):
+        """Create a new AnnealingResultData instance.
+
+        Args:
+            job_data (Optional[dict[str, Any]], optional):
+                Ex: [{'spin': {' x1': 0, ' x2': 0, 'x1': 0},\
+                'energy': 0, 'time': 0.006517000030726194,\
+                'constraint': True, 'memory_usage': 1.189453125}]. 
+                Defaults to None.
+        """
+        super().__init__(**kwargs)
+        self._job_data = job_data if job_data else {}
+
+    @property
+    def experiment_type(self) -> ExperimentType:
+        """Returns the experiment type."""
+        return ExperimentType.ANNEALING
+
+    @classmethod
+    def from_object(cls, model: AnnealingExperimentMetadata, **kwargs) -> AnnealingResultData:
+        """Creates a new AnnealingResultData instance from a AnnealingExperimentMetadata object."""
+        return cls.from_dict(model.model_dump(**kwargs))
+
+    @classmethod
+    def from_dict(cls, data: dict[str, Any]) -> AnnealingResultData:
+        """Creates a new AnnealingResultData instance from a dictionary."""
+        return cls(job_data=data.pop("job_data", {}))
+
+    def to_dict(self) -> dict[str, Any]:
+        """Converts the AnnealingResultData instance to a dictionary."""
+        return {
+            "job_data": self._job_data,
+        }
+
+    def get_annealing_results(self) -> dict[str, Any]:
```

Comment:
Can probably just rename to `get_results()`. `AnnealingResultData.get_annealing_results()` is a bit redundant